File: go/src/leetcode/editor/cn/0414_third_maximum_number.go
package main

import "sort"

func main() {
	println("414、third_maximum_number")
}

// leetcode submit region begin(Prohibit modification and deletion)
func thirdMax(nums []int) int {
	sort.Ints(nums)
	res, count := 0, 0
	for i := len(nums) - 1; i >= 0; i-- {
		if i == len(nums)-1 || nums[i] != nums[i+1] {
			res = nums[i]
			count++
		}
		if count == 3 {
			return res
		}
	}
	return nums[len(nums)-1]
}

//leetcode submit region end(Prohibit modification and deletion)

File: rust/src/leetcode/editor/cn/_0414_third_maximum_number.rs
#[cfg(test)]
mod tests {
    #[allow(unused_imports)]
    use super::*;

    #[test]
    fn test() {
        println!("414、third_maximum_number");
    }
}

#[allow(dead_code)]
struct Solution;

//leetcode submit region begin(Prohibit modification and deletion)
impl Solution {
    #[allow(dead_code)]
    pub fn third_max(mut nums: Vec<i32>) -> i32 {
        nums.sort_unstable();
        let mut res = 0;
        let mut mark = 0;
        for i in (0..nums.len()).rev() {
            if i == nums.len() - 1 || nums[i] != nums[i + 1] {
                res = nums[i];
                mark += 1;
            }
            if mark == 3 {
                return res;
            }
        }
        return nums[nums.len() - 1];
    }
}
//leetcode submit region end(Prohibit modification and deletion)
